core/src/main/scala/kafka/zk/AdminZkClient.scala

```diff
@@ -125,6 +125,11 @@ class AdminZkClient(zkClient: KafkaZkClient) extends Logging {
         throw new InvalidReplicaAssignmentException("Duplicate replica assignment found: " + partitionReplicaAssignment)
     )
 
+    val partitionSize = partitionReplicaAssignment.size
+    val sequenceSum = partitionSize * (partitionSize - 1) / 2
```

Comment:
I am not sure if this validation covers everything. For example a partition list of (0, 0, 3) has the same sum of (0, 1, 2) and the former should be invalid.

Reply:
@junrao, I add a duplication check and non-negative check in  #275727a, so invalid partitions like (0,0,3) or (-1,1,2,4) won't pass anymore, that should cover all edge cases.